components/mongodb/mongodb.app.mjs

```diff
@@ -1,11 +1,234 @@
+                strict: false,
+              });
+              const Model = mongoose.model(collection, schema);
+              const document = await Model.findById(_id);
+              mongoose.connection.close(() => {
+                setTimeout(() => resolve(document));
+              });
+            } catch (err) {
+              mongoose.connection.close(() => {
+                setTimeout(() => reject(err));
+              });
+            }
+          })
+          .catch((err) => reject(err));
+      });
+    },
+    searchDocuments(collection, filter) {
+      return new Promise((resolve, reject) => {
+        this.connect()
+          .then(async () => {
+            try {
+              const schema = new mongoose.Schema(undefined, {
+                strict: false,
+              });
+              const Model = mongoose.model(collection, schema);
+              const documents = await Model.find(filter);
+              mongoose.connection.close(() => {
+                setTimeout(() => resolve(documents));
+              });
+            } catch (err) {
+              mongoose.connection.close(() => {
+                setTimeout(() => reject(err));
+              });
+            }
+          })
+          .catch((err) => reject(err));
+      });
+    },
+    getSchemaByData(data) {
+      const keys = Object.keys(data);
+      const schemaFields = {};
+      for (let i = 0; i < keys.length; i++) {
+        schemaFields[keys[i]] = {
+          type: mongoose.Schema.Types.Mixed,
+        };
+      }
+
+      const schema = new mongoose.Schema(schemaFields, {
+        strict: false,
+      });
+
+      return schema;
+    },
+    parseStrings(dataParam, parseNumbers, parseBooleans, parseDates) {
+      const data = {
+        ...dataParam,
+      };
+
+      const keys = Object.keys(data);
+      for (let i = 0; i < keys.length; i++) {
+        if (parseBooleans && this.isBooleanString(data[keys[i]])) {
+          data[keys[i]] = this.parseBoolean(data[keys[i]]);
+          continue;
+        }
+
+        if (parseDates && this.isDate(data[keys[i]])) {
+          data[keys[i]] = new Date(data[keys[i]]);
+          continue;
+        }
+
+        if (parseNumbers && !isNaN(data[keys[i]])) {
+          data[keys[i]] = parseFloat(data[keys[i]]);
+          continue;
+        }
+      }
+
+      return data;
+    },
+    parseBoolean(string) {
+      switch (string.toLowerCase().trim()) {
+      case "true":
+      case "1":
+      case 1:
+        return true;
+
+      case "false":
+      case "0":
+      case 0:
+        return false;
+      }
+    },
+    isBooleanString(string) {
+      switch (string.toLowerCase().trim()) {
+      case "true":
+      case "1":
+      case 1:
+      case "false":
+      case "0":
+      case 0:
+        return true;
+
+      default:
+        return false;
+      }
+    },
+    isDate(str) {
+      return !isNaN(new Date(str).getDate());
     },
```

Comment:
We need a different way to determine if the value is a date. This is returning true for any number. If I enter the value as 4, it's entered in the database as "2001-04-01T00:00:00.000+00:00".

Reply:
Humm, okay, I will check it